### PR TITLE
Migrate to npe2

### PR DESCRIPTION
**Pull request by jamesyan-git**
This PR adds the necessary metadata to convert npe2 and removes mentions of hook implementations.
Additionally moved all imports form `plugin.py` to `init` and kept reader separate (now found in `reader.py`)

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -52,7 +52,12 @@ def read(fname):
...
-                'console_scripts': ['zarpaint = zarpaint._main:main']
-                },
+             "napari.manifest": [
+                 "zarpaint = zarpaint:napari.yaml",
+             ],
+             'console_scripts': [
+                 'zarpaint = zarpaint._main:main'
+            ]
+         },
+         package_data={"zarpaint": ["napari.yaml"]},
```
Comment (jni) [inline setup.py:62]: Ah. Interesting — would you mind migrating this to setup.cfg while you're at it? The reason it's still setup.py is only historical.

File: zarpaint/__init__.py [hunk 2]
```diff
@@ -6,11 +6,12 @@
 from ._zarpaint import create_labels, open_tensorstore
 from ._dims_chooser import DimsSorter, set_axis_labels
 from ._watershed import watershed_split
-
+from ._add_3d_points import add_points_3d_with_alt_click
```
Comment (jni) [inline zarpaint/__init__.py:9]:
```suggestion
from ._add_3d_points import add_points_3d_with_alt_click

```

File: zarpaint/napari.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    title: Create add_points_3d_with_alt_click
+    python_name: zarpaint:add_points_3d_with_alt_click
+  - id: zarpaint.set_axis_labels
+    title: Create set_axis_labels
+    python_name: zarpaint:set_axis_labels
+  - id: zarpaint.DimsSorter
+    title: Create Dims Sorter
+    python_name: zarpaint:DimsSorter
+  - id: zarpaint.create_labels
+    title: Create create_labels
```
Comment (jni) [inline zarpaint/napari.yaml:15]:
```suggestion
    title: Create labels...
```

File: zarpaint/napari.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+name: zarpaint
+schema_version: 0.1.0
+contributions:
+  commands:
+  - id: zarpaint.add_points_3d_with_alt_click
+    title: Create add_points_3d_with_alt_click
```
Comment (jni) [inline zarpaint/napari.yaml:6]:
```suggestion
    title: Add points with alt-click in 3D
```

File: zarpaint/napari.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    python_name: zarpaint:add_points_3d_with_alt_click
+  - id: zarpaint.set_axis_labels
+    title: Create set_axis_labels
```
Comment (jni) [inline zarpaint/napari.yaml:9]:
```suggestion
    title: Set axis labels...
```

File: zarpaint/napari.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    python_name: zarpaint:set_axis_labels
+  - id: zarpaint.DimsSorter
+    title: Create Dims Sorter
```
Comment (jni) [inline zarpaint/napari.yaml:12]:
```suggestion
    title: Sort dims...
```

File: zarpaint/napari.yaml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    python_name: zarpaint:DimsSorter
+  - id: zarpaint.create_labels
+    title: Create create_labels
+    python_name: zarpaint:create_labels
+  - id: zarpaint.watershed_split
+    title: Create watershed_split
```
Comment (jni) [inline zarpaint/napari.yaml:18]:
```suggestion
    title: Split with watershed
```

As above, this doesn't actually need the widget I think.

File: zarpaint/napari.yaml [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    python_name: zarpaint:watershed_split
+  - id: zarpaint.get_reader
+    title: Get Reader
+    python_name: zarpaint.reader:zarr_tensorstore
```
Comment (jni) [inline zarpaint/napari.yaml:22]:
```suggestion
```

Do both readers and widgets need actions?

File: zarpaint/napari.yaml (hunk 4 above)
Comment (jni) [reply zarpaint/napari.yaml:6]: Wait, I just noticed that these are commands, not widgets. Can npe2 already add parameter-less function calls without a widget? If so, can you convert this to just a command that runs without the widget?

**Review by jni — CHANGES_REQUESTED** [review]
Interesting! This might need more input for @dragadoncila as I don't know how far along npe2 is at the moment. Either way, I prefer "Action..." as the menu bar title for an item to "Create Action".

File: zarpaint/napari.yaml (hunk 4 above)
Comment (DragaDoncila) [reply zarpaint/napari.yaml:6]: Sadly, no, you still need the widget. The `command` contribution is associated with the `widget` contribution via the `command_id`.

**Comment by DragaDoncila** [comment]
> Either way, I prefer "Action..." as the menu bar title for an item to "Create Action".

@jni just to clarify. The `title` of a command is what would show up if you were to say, search for it through a VSCode like command palette. So they should stay stuff like "Open segmentation widget". The `display_name` of a `Widget` contribution is what comes up in the menu bar, so it should say something like "Segment with watershed". 

Eventually the `command`s might stand alone to perform other actions, and I feel like this will make more sense then.

File: zarpaint/napari.yaml (hunk 8 above)
Comment (DragaDoncila) [reply zarpaint/napari.yaml:22]: Yep, at this stage you need both a `command` contribution and an associated `reader`/`widget` contribution.

**Comment by jni** [comment]
@DragaDoncila thanks for the clarifications!

> So they should stay stuff like "Open segmentation widget".

I actually disagree with this. I think it should *still* say "Segment with watershed…". No one is using napari to open widgets, they are using it to achieve specific visualisation, annotation, and analysis goals, and the commands should be named in a way that guides users towards those goals.

The "…" is important as it signifies that this command will take further input, see [Apple's Human Interface Guidelines on menu item titles](https://developer.apple.com/design/human-interface-guidelines/macos/menus/menu-anatomy/#menu-and-menu-item-titles). I guess for now all commands have an ellipsis. Once direct commands are supported, we can remove the ellipses where needed.

> Use an ellipsis whenever choosing a menu item requires additional input from the user. The ellipsis character (…) means a dialog or separate window will open and prompt the user for additional information or to make a choice.

I don't see a need to distinguish commands from the command palette and menu bar items in this regard: both cause the same effect.

As an aside, we should probably codify this in npe2 plugin best practices somewhere...

Based on this I'll push some preferred menu item titles directly to this branch and merge. Thanks @jamesyan-git! Onward and upward! :smiley:

**Pull request merged**